src/Kiota.Builder/Writers/HTTP/CodeClassDeclarationWriter.cs

```diff
@@ -0,0 +1,365 @@
+    private static void WriteProperties(CodeClass requestBodyClass, LanguageWriter writer)
+    {
+        var properties = requestBodyClass.Properties.Where(prop => prop.IsOfKind(CodePropertyKind.Custom)).ToList();
+        for (int i = 0; i < properties.Count; i++)
+        {
+            var prop = properties[i];
+            var propName = $"\"{prop.Name}\"";
+            writer.Write($"{propName}: ");
+            if (prop.Type is CodeType propType && propType.TypeDefinition is CodeClass propClass)
+            {
+                // If the property is an object, write a JSON representation recursively
+                writer.WriteLine("{", includeIndent: false);
+                writer.IncreaseIndent();
+                WriteProperties(propClass, writer);
+                writer.DecreaseIndent();
+                writer.Write("}");
+            }
+            else
+            {
+                writer.Write(GetDefaultValueForProperty(prop), includeIndent: false);
+            }
+
+            // Add a trailing comma if there are more properties to be written
+            if (i < properties.Count - 1)
+            {
+                writer.WriteLine(",", includeIndent: false);
+            }
+            else
+            {
+                writer.WriteLine();
+            }
+        }
+
+        // If the class extends another class, write properties of the base class
+        if (requestBodyClass.StartBlock.Inherits?.TypeDefinition is CodeClass baseClass)
+        {
+            WriteProperties(baseClass, writer);
+        }
+    }
+
+    /// <summary>
+    /// Gets the default value for the given property.
+    /// </summary>
+    /// <param name="codeProperty">The property to get the default value for.</param>
+    /// <returns>The default value as a string.</returns>
+    private static string GetDefaultValueForProperty(CodeProperty codeProperty)
+    {
+        return codeProperty.Type.Name switch
+        {
+            "int" or "integer" => "0",
+            "string" => "\"string\"",
+            "bool" or "boolean" => "false",
+            _ when codeProperty.Type is CodeType enumType && enumType.TypeDefinition is CodeEnum enumDefinition =>
+                enumDefinition.Options.FirstOrDefault()?.Name is string enumName ? $"\"{enumName}\"" : "null",
+            _ => "null"
+        };
+    }
+
+    private static string BuildUrlStringFromTemplate(string urlTemplateString, List<CodeProperty> queryParameters, List<CodeProperty> pathParameters, string? baseUrl)
+    {
+        // Use the provided baseUrl or default to "http://localhost/"
+        baseUrl ??= "http://localhost/";
+
+        // unquote the urlTemplate string and replace the {+baseurl} with the actual base url string
+        urlTemplateString = urlTemplateString.Trim('"').Replace("{+baseurl}", baseUrl, StringComparison.InvariantCultureIgnoreCase);
+
+        // Build RequestInformation using the URL
+        var requestInformation = new RequestInformation()
```

Comment:
I believe he's using the `RequestInformation` resolve a valid url from the provided query and path parameters. So that the placeholders can be filled rather than have placeholders(which is what the uritemplate has)